Source/WebCore/Modules/mediasession/MediaMetadata.cpp

```diff
@@ -159,22 +161,101 @@ ExceptionOr<void> MediaMetadata::setArtwork(ScriptExecutionContext& context, Vec
     return { };
 }
 
+// We attempt to give a score to the image dimensions. This score is between 0 and 1.
+// A negative score indicates an invalid image
+// A score of 0 is for images smaller than the minimum size
+// A score of 1 indicates an image of ideal size with an aspect ratio of 1 (square)
+// The closer to the ideal size, the higher the score.
+float MediaMetadata::imageDimensionsScore(int width, int height) const
+{
+    IntSize size { width, height };
+    if (size.isEmpty())
+        return -1;
+
+    if (size.maxDimension() <= m_minimumSize)
+        return 0; // Ignore images that are too small.
+
+    // We account for the aspect ratio in scoring the "best" artwork.
+    // We prefer artwork's images with a square AR.
+    double longEdge = size.maxDimension();
+    double shortEdge = size.minDimension();
+    auto aspectRatioCoefficient = shortEdge / longEdge;
+
+    if (longEdge < m_idealSize)
+        return aspectRatioCoefficient * (0.8 * (longEdge - m_minimumSize) / (m_idealSize - m_minimumSize) + 0.2);
```

Comment:
Theoretically, it would be possible for m_idealSize and m_minimumSize to be the same.
Given they are static, we should probably rename them as s_...
Or maybe make this method a free function and have const variables for ideal/minimum sizes.

Reply:
I added a static_assert